src/test/java/net/snowflake/client/jdbc/ConnectionLatestIT.java

```diff
@@ -214,7 +214,7 @@ public void putGetStatementsHaveQueryIDEvenWhenFail() throws Throwable {
         statement.executeQuery("PUT file://not_existing_file @" + stageName);
         fail("PUT statement should fail");
       } catch (Exception __) {
-        assertNull(snowflakeStatement.getQueryID());
+        TestUtil.assertValidQueryId(snowflakeStatement.getQueryID());
```

Comment:
Does it makes sense to check here that exactly the same queryId is provided by the exception?
There are also 3 new tests in StatementLatestIT eg. `testQueryIdIsSetOnFailedExecuteQuery` were tests showing than those two (exception and statement) should provide the same QueryId.

Reply:
I added exception query id checks